sema/check/check.py

```
@@ -0,0 +1,66 @@
+            "Check service initialized with input_folder: %s, output: %s",
+            self.input_folder,
+            self.output,
+        )
+
+        assert self.input_folder, "input_folder not provided"
+        assert self.output, "output not provided"
+
+        self._result = CheckResult()
+
+    @Trace.init(Trace)
+    def process(self) -> None:
```

Ruff

Missing type annotation for `self` in method

(ANN101)